### DISCUSSION TITLE
website without www not working

### DISCUSSION THREAD

**Discussion by sheminasalam** [w35l3y/hassio-addons#5] [General]
Hi there, I really appreciate your work. I have started using this addon recently only and it has been working fine. I think I need help with configuring cloudflare dns. I am able to page loaded with ```www.example.com``` but not with just ```example.com``` . I have tried adding a CNAME record in cloudflare pointing root to ```www.example.com``` and also even to ```tunnel_id..cfargotunnel.com```. Both doesnot work and there is also a warning CNAME ```Because CNAME records are not allowed at the zone apex (RFC 1034), CNAME flattening will be applied to this record.```

What am I missing here?

**Comment by w35l3y** [comment]
Good question. I don't really know.
You should ask cloudflare support or search on the internet.
It is beyond of what the addon does.

**Reply by sheminasalam** [reply]
I found the solution, hope you can add this to your documentation.

After adding the cname for www we also have to another cname with ```example.com``` as the name and the same tunnel_id..cfargotunnel.com as the target. It is similar to www cname just the name is different.

Then we have to set a rule. Go to rules tab and make a rule to forward url ```example.com``` to ```https://www.example.com``` . Save and deploy, thats all

**Reply by w35l3y** [reply]
I didn't understand.

Could you send screenshots of all you had to do ?
If so, don't forget to ommit sensitive data.

**Reply by sheminasalam** [reply]
First we have to set the cname config like below.

![FireShot Capture 524 - DNS - therivana21 online - Account - Cloudflare - Web Performance   S_ - dash cloudflare com](https://user-images.githubusercontent.com/45196177/142648146-bf2ecad0-5088-40ed-b460-cf7a4e095537.png)

The go to the rules page in cloudflare and click create page rule and then fill the details like below.
![FireShot Capture 526 - Rules - therivana21 online - Account - Cloudflare - Web Performance  _ - dash cloudflare com](https://user-images.githubusercontent.com/45196177/142648318-5bed3c91-4e7a-45e0-bc63-e6ec0a0160a3.png)

Click save and deploy and you will see the new rule like below in the rules page.

![FireShot Capture 525 - Rules - therivana21 online - Account - Cloudflare - Web Performance  _ - dash cloudflare com](https://user-images.githubusercontent.com/45196177/142648458-aed817de-2cb7-4677-9f0f-5492c1bff7d8.png)

**Reply by w35l3y** [reply]
As I thought, it is in the cloudflare side. It is out of the scope of the addon.

**Comment by sheminasalam** [comment]
Not connected to this but an issue the github url in addon page is not valid. It is ```https://github.com/w35l3y/hassio-addons/cloudflare_tunnel```. Hope next update could change it also
